File: src/main/java/diff/patch/accessors/MapSetter.java
package diff.patch.accessors;

import diff.patch.Setter;

import java.util.Map;

/**
 *
 * @author sebasjm <smarchano@primary.com.ar>
 */
public class MapSetter<Key, Value> implements Setter<Value> {
    final Map<Key, Value> target;
    final Key field;

    public MapSetter(Map<Key, Value> target, Key field) {
        this.target = target;
        this.field = field;
    }

    @Override
    public void set(Value value) {
        target.put(field, value);
    }
    
}

File: src/main/java/diff/patch/accessors/MapSetter.kt
package diff.patch.accessors

import diff.patch.Setter

/**

 * @author sebasjm <smarchano></smarchano>@primary.com.ar>
 */
class MapSetter<Key: Any, Value: Any>(internal val target: MutableMap<Key, Value>, internal val field: Key) : Setter<Value> {

    override fun set(value: Value) {
        target.put(field, value)
    }

}
